dask/dataframe/core.py

```diff
@@ -3228,25 +3229,27 @@ def astype(self, dtype):
             M.astype, dtype=dtype, meta=meta, enforce_metadata=False
         )
 
-    @derived_from(pd.Series)
-    def append(self, other, interleave_partitions=False):
-        if PANDAS_GT_140:
-            warnings.warn(
-                "The frame.append method is deprecated and will be removed from"
-                "dask in a future version. Use dask.dataframe.concat instead.",
-                FutureWarning,
-            )
-        # because DataFrame.append will override the method,
-        # wrap by pd.Series.append docstring
-        from dask.dataframe.multi import concat
+    if not PANDAS_GT_200:
 
-        if isinstance(other, (list, dict)):
-            msg = "append doesn't support list or dict input"
-            raise NotImplementedError(msg)
+        @derived_from(pd.Series)
+        def append(self, other, interleave_partitions=False):
+            if PANDAS_GT_140:
+                warnings.warn(
+                    "The frame.append method is deprecated and will be removed from"
+                    "dask in a future version. Use dask.dataframe.concat instead.",
+                    FutureWarning,
+                )
```

Comment:
```suggestion
            warnings.warn(
                "The frame.append method is deprecated and will be removed from"
                "dask in a future version. Use dask.dataframe.concat instead.",
                FutureWarning,
            )
```

Reply:
Why this change? Currently `dask.dataframe` and `pandas` have this warning synced

Reply:
The method will be removed from a future version of dask.dataframe for all users, including those of pandas < 1.4. The condition makes no sense to me.

Reply:
It so happened that @crusaderky and I ended up chatting about this offline. He said he was okay with the approach in this PR.